src/FlowtideDotNet.Substrait/Sql/Internal/SqlSubstraitVisitor.cs

```diff
@@ -95,27 +132,125 @@ public SqlSubstraitVisitor(SqlPlanBuilder sqlPlanBuilder, SqlFunctionRegister sq
+                    else if (upperName == "DISTRIBUTED")
+                    {
+                        var upperVal = opt.Value.ToSql().ToUpper();
+                        if (upperVal == "TRUE")
+                        {
+                            isDistributed = true;
+                        }
+                    }
+                    else if (upperName == "SCATTER_BY")
+                    {
+                        if (opt.Value is Value.StringBasedValue stringBasedVal)
+                        {
+                            var exprVisitor = new SqlExpressionVisitor(sqlFunctionRegister);
+                            // Do a lookup on the partition by field
+                            var exprData = exprVisitor.Visit(new Expression.Identifier(new Ident(stringBasedVal.Value)), relationData.EmitData);
+                            if (exprData.Expr is Expressions.FieldReference fieldReference)
+                            {
+                                scatterField = fieldReference;
+                            }
+                            else
+                            {
+                                throw new InvalidOperationException("SCATTER_BY epxects a field reference.");
+                            }
+                        }
+                        else
+                        {
+                            throw new InvalidOperationException("SCATTER_BY expects a string based value with qoutes.");
+                        }
+                    }
+                    else if (upperName == "PARTITION_COUNT")
+                    {
+                        if (int.TryParse(opt.Value.ToSql(), out var partitionCountValue))
+                        {
+                            partitionCount = partitionCountValue;
+                        }
+                        else
+                        {
+                            throw new InvalidOperationException($"Invalid partition count expected a number got '{opt.Value.ToSql()}'");
+                        }
+                    }
+                    else
+                    {
+                        throw new InvalidOperationException($"Unknown option '{opt.Name}' in create view statement");
                     }
                 }
             }
 
             var relation = relationData.Relation;
 
+            var viewName = createView.Name.ToSql();
+
             if (isBuffered)
             {
                 relation = new BufferRelation()
                 {
                     Input = relation
                 };
             }
+            if (isDistributed)
+            {
+                ExchangeKind? exchangeKind = default;
+                if (scatterField != null)
+                {
+                    exchangeKind = new ScatterExchangeKind()
+                    {
+                        Fields = new List<Expressions.FieldReference>()
+                        {
+                            scatterField
+                        }
```

Comment:
```suggestion
Fields = [scatterField]
```
could be changed to this. It's available at a few other places too but I just leave it here as. It's a style change I guess, so up to you what you like most :)